doc/result.md

```diff
@@ -246,6 +246,50 @@ Result.Success<int, string>(100).Transpose();
 Result.Failure<int, string>("message").Transpose();
 ```
 
+## Working with `Result<Option<T>, TFailure>`
+
+### Match (Overload)
```

Comment:
My editor was complaining about duplicate header text, hence the `(Overload)` descriptor